nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPath.java

```diff
@@ -255,6 +259,78 @@ public static AccessPath getAccessPathForNodeWithMapGet(Node node, @Nullable Typ
     }
   }
 
+  /**
+   * Constructs an access path ending with the class field element in the argument. The receiver is
+   * the method receiver itself.
+   *
+   * @param element the receiver element.
+   * @return access path representing the class field
+   */
+  public static AccessPath fromElement(Element element) {
+    assert element.getKind().isField()
+        : "element must be of type: FIELD but received: " + element.getKind();
+    Root root = new Root();
+    return new AccessPath(root, Collections.singletonList(new AccessPathElement(element)));
+  }
+
+  /**
+   * Extends an access path with a class field element.
+   *
+   * @param receiverNode the receiver node
+   * @param field element of the class field
+   * @return the extended access path
+   */
+  public static @Nullable AccessPath extendReceiverNodeAccessPathWithField(
+      @Nonnull Node receiverNode, Element field) {
+    if (receiverNode.getTree() == null) {
+      // "this" is the receiver.
+      return fromElement(field);
+    }
+    return extendReceiverTreeAccessPathWithField(receiverNode.getTree(), field);
+  }
+
+  /**
+   * Extends an access path with a class field element.
+   *
+   * @param receiverTree the receiver tree
+   * @param field element of the class field
+   * @return the extended access path
+   */
+  public static @Nullable AccessPath extendReceiverTreeAccessPathWithField(
+      @Nonnull Tree receiverTree, Element field) {
+    if (receiverTree.getKind().equals(Tree.Kind.CONDITIONAL_EXPRESSION)) {
```

Comment:
I don't think we should blacklist conditional expressions.  There may be other cases we cannot handle (e.g., chains of method calls with arguments).  Instead, we should do a case analysis on the type of `Tree` and its subtrees and only handle the valid cases (like `MemberSelectTree`, `IdentifierTree`, etc.).

Reply:
The logic here should be similar to `populateElementsRec` (which unfortunately does not have Javadoc):

https://github.com/uber/NullAway/blob/cd93d7fd90f4a13f05cd49f5961ebac3193afbd2/nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPath.java#L265

That method recurses through a tree structure, handling valid cases, and returns `null` if a case cannot be handled.  That `null` "bubbles up" to the top-level return value, which indicates to the top-level caller whether the entire expression could be handled or not.  Our logic here should be similar, and we should probably factor out some overlapping logic (particularly related to constant arguments to method calls).

Reply:
Changed it to the following (If I understood your preferred style correctly):

```java
switch (receiverTree.getKind()){
      case MEMBER_SELECT:
      case IDENTIFIER:
        List<AccessPathElement> receivers = getReceiverAccessPathElementChain(receiverTree);
        receivers.add(new AccessPathElement(field));
        Root root = new Root(receivers.get(0).getJavaElement());
        accessPath = new AccessPath(root, receivers.subList(1, receivers.size()));
        break;
      default:
        accessPath = null;
}
```